config/config.go

```diff
@@ -8,21 +8,25 @@ import (
 	"github.com/jessevdk/go-flags"
 	"github.com/pkg/errors"
 	"os"
-	"reflect"
 )
 
 // ErrInvalidArgument is the error returned by [ParseFlags] or [FromYAMLFile] if
 // its parsing result cannot be stored in the value pointed to by the designated passed argument which
 // must be a non-nil pointer.
 var ErrInvalidArgument = stderrors.New("invalid argument")
 
+// validatorPtrTo combines the [Validator] interface with a pointer constraint.
+type validatorPtrTo[T any] interface {
+	Validator
+	*T
+}
+
 // FromYAMLFile parses the given YAML file and stores the result
 // in the value pointed to by v. If v is nil or not a pointer,
 // FromYAMLFile returns an [ErrInvalidArgument] error.
-func FromYAMLFile(name string, v Validator) error {
-	rv := reflect.ValueOf(v)
-	if rv.Kind() != reflect.Pointer || rv.IsNil() {
-		return errors.Wrapf(ErrInvalidArgument, "non-nil pointer expected, got %T", v)
+func FromYAMLFile[T any, V validatorPtrTo[T]](name string, v V) error {
+	if v == nil {
+		return errors.Wrap(ErrInvalidArgument, "got nil pointer")
```

Comment:
While on it, we could also remove those right away.

Reply:
Why should we allow passing nil (pointers) and why is it an argument that this is not checked by other functions? If we omit this check, `defaults.Set()` in `FromYAMLFile()` will panic and `ParseFlags()` will return different errors depending on whether flags were specified on the command line or not (all if nil was specified). So, why?

Reply:
It's not only our functions, but also the Go stuff doesn't nil check pointer params, e.g. all methods of atomic.Int64, because those are programming errors. IMAO [Pete explained](https://softwareengineering.stackexchange.com/a/298486) it very well.

Reply:
> e.g. all methods of atomic.Int64

So you are comparing low-level pointer receiver functions with our high level library functions?

> because those are programming errors

True. But why is it a problem to just indicate that with a clear error message in a library? Besides, `ParseFlags()` **does not** panic. It yields random errors on first glance.

>IMAO [Pete explained](https://softwareengineering.stackexchange.com/a/298486) it very well.

Why can't you just summarize it? Why should I read this wall of text? Besides, I would prefer your own arguments to those of a random person who has  been "using [Go] professionally for about 6 months". 🤣

> It's not only our functions, but also the Go stuff doesn't nil check pointer params

Similar functions like `json.Unmarshal()`, `errors.As()`, etc. are a perfect example of how it's done.